presto-main/src/main/java/com/facebook/presto/operator/scalar/DataSizeFunctions.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.scalar;
+
+import com.facebook.presto.spi.PrestoException;
+import com.facebook.presto.spi.function.Description;
+import com.facebook.presto.spi.function.LiteralParameters;
+import com.facebook.presto.spi.function.ScalarFunction;
+import com.facebook.presto.spi.function.SqlType;
+import io.airlift.slice.Slice;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+
+import static com.facebook.presto.spi.StandardErrorCode.INVALID_FUNCTION_ARGUMENT;
+import static com.facebook.presto.spi.StandardErrorCode.NUMERIC_VALUE_OUT_OF_RANGE;
+import static com.facebook.presto.spi.type.Decimals.encodeUnscaledValue;
+import static java.lang.Character.isDigit;
+import static java.lang.String.format;
+
+public class DataSizeFunctions
+{
+    private DataSizeFunctions() {}
+
+    @Description("converts data size string to bytes")
+    @ScalarFunction("parse_data_size")
+    @LiteralParameters("x")
+    @SqlType("decimal(38,0)")
+    public static Slice parseDataSize(@SqlType("varchar(x)") Slice input)
+    {
+        String dataSize = input.toStringUtf8();
+
+        int valueLength = 0;
+        for (int i = 0; i < dataSize.length(); i++) {
+            char c = dataSize.charAt(i);
+            if (isDigit(c) || c == '.') {
+                valueLength++;
+            }
+            else {
+                break;
+            }
+        }
+
+        checkDataSize(valueLength > 0, dataSize);
+        String valueString = dataSize.substring(0, valueLength);
+        BigDecimal value = parseValue(valueString, dataSize);
+        String unitString = dataSize.substring(valueLength);
+        Unit unit = parseUnit(unitString, dataSize);
+
+        BigInteger bytes = value.multiply(unit.getFactor()).toBigInteger();
+        try {
+            return encodeUnscaledValue(bytes);
+        }
+        catch (ArithmeticException e) {
+            throw new PrestoException(NUMERIC_VALUE_OUT_OF_RANGE, format("Value out of range: '%s' ('%sB')", dataSize, bytes));
+        }
+    }
+
+    private static BigDecimal parseValue(String value, String dataSize)
+    {
+        try {
+            return new BigDecimal(value);
+        }
+        catch (NumberFormatException e) {
+            throw invalidDataSize(dataSize);
+        }
+    }
+
+    private static Unit parseUnit(String unitString, String dataSize)
+    {
+        if (unitString.isEmpty()) {
```

Comment:
This should probably be an error. I don't think we want to accept `123` as a data size (should be `123B`)